eth/backend.go

```diff
@@ -683,18 +689,25 @@ func (s *Ethereum) startCheckpointWhitelistService() {
 		fnName         = "whitelist checkpoint"
 	)
 
-	s.retryHeimdallHandler(s.handleWhitelistCheckpoint, tickerDuration, whitelistTimeout, fnName)
+	s.retryHeimdallHandler(s.fetchAndHandleWhitelistCheckpoint, tickerDuration, whitelistTimeout, fnName)
 }
 
 // startMilestoneWhitelistService starts the goroutine to fetch milestiones and update the
 // milestone whitelist map.
 func (s *Ethereum) startMilestoneWhitelistService() {
-	const (
-		tickerDuration = 12 * time.Second
-		fnName         = "whitelist milestone"
-	)
+	ethHandler, bor, _ := s.getHandler()
 
-	s.retryHeimdallHandler(s.handleMilestone, tickerDuration, whitelistTimeout, fnName)
+	// If heimdall ws is available use WS subscription to new milestone events instead of polling
+	if hmm.IsHeimdallV2 && bor != nil && bor.HeimdallWSClient != nil {
+		s.subscribeAndHandleMilestone(context.Background(), ethHandler, bor)
```

Comment:
Shall we handle the error here?

Reply:
Added error handling and also on error fallback to HTTP.